app/grandchallenge/cases/image_builders/tiff.py

```diff
@@ -214,52 +250,196 @@ def _add_folder_uploads(*, dzi_output, image, new_folder_upload):
     return new_folder_upload
 
 
-def image_builder_tiff(path: Path, session_id=None) -> ImageBuilderResult:
+mirax_pattern = r"INDEXFILE\s?=\s?.|FILE_\d+\s?=\s?."
+# vms (and vmu) files conatin key value pairs, where the ImageFile keys
+# can have the following format:
+# ImageFile =, ImageFile(x,y) ImageFile(z,x,y)
+vms_pattern = r"ImageFile(\(\d*,\d*(,\d)?\))?\s?=\s?.|MapFile\s?=\s?.|OptimisationFile\s?=\s?.|MacroImage\s?=\s?."
+
+
+def _compile_mrx(files: List[Path], converter) -> List[GrandChallengeTiffFile]:
+    def get_filenames_from_ini(ini_file: Path):
+        file_matched = []
+        if not ini_file.exists():
+            ini_file = [
+                f
+                for f in ini_file.parent.iterdir()
+                if f.name.lower() == ini_file.name.lower()
+            ][0]
+        with open(ini_file, "r") as f:
+            lines = [
+                line for line in f.readlines() if re.match(mirax_pattern, line)
+            ]
+            for line in lines:
+                original_name = line.split("=")[1].strip()
```

Comment:
Could there be lines without `=` in them? E.g. an empty line at the end of the file? In that case it might be safe to wrap this in a `try.. except IndexError`.

Reply:
No, not possible actually, the regex makes sure of that.